api/groups.go

```diff
@@ -20,21 +20,32 @@ func (api *API) CreateGroupHandler(ctx context.Context, w http.ResponseWriter, r
 		return nil, handleBodyReadError(ctx, err)
 	}
 
-	createGroup := models.CreateGroup{}
+	createGroup := models.CreateUpdateGroup{}
 	// GroupName is UUID
-	createGroup.GroupName = uuid.NewString()
+	uuid := uuid.NewString()
+	createGroup.GroupName = &uuid
 	err = json.Unmarshal(body, &createGroup)
 	if err != nil {
 		return nil, handleBodyUnmarshalError(ctx, err)
 	}
 
-	validationErrs := createGroup.ValidateCreateGroupRequest(ctx)
+	createGroup.GroupsList, err = api.GetListGroups()
+	if err != nil {
+		cognitoErr := models.NewCognitoError(ctx, err, "Cognito ListGroups request from update group endpoint")
+		if cognitoErr.Code == models.NotFoundError {
+			return nil, models.NewErrorResponse(http.StatusNotFound, nil, cognitoErr)
+		}
+		return nil, models.NewErrorResponse(http.StatusInternalServerError, nil, cognitoErr)
+	}
+
+	validationErrs := createGroup.ValidateCreateUpdateGroupRequest(ctx)
 	if len(validationErrs) != 0 {
 		return nil, models.NewErrorResponse(http.StatusBadRequest, nil, validationErrs...)
 	}
 
-	// build create group input
+	// remove special chars from description
 	createGroup.CleanGroupDescription()
```

Comment:
Isn't this already run in the validation step?
(although see comment in the validation step about if this is needed)

Reply:
The `CleanGroupDescription()` function is applied to the incoming data after the validation step - as calling it before the validation, and without checking if the description is populated, would result in a null pointer exception.